window-sanity.h still appears inccorect

does not define ERROR at all if not previously seen, but code will still try to reference the win32::ERROR entity.
(https://github.com/capnproto/capnproto/blob/55a368beb987abf9eeb9b3843e9c5423ad37ab29/c%2B%2B/src/kj/windows-sanity.h#L52-L56)

Perhaps applying a patch similar to following would be an improvement:
```
diff --git a/c++/src/kj/windows-sanity.h b/c++/src/kj/windows-sanity.h
index 894ec1d7..f18b39e6 100644
--- a/c++/src/kj/windows-sanity.h
+++ b/c++/src/kj/windows-sanity.h
@@ -49,9 +49,13 @@
 #pragma once
 
 namespace win32 {
+#ifdef ERROR  // This could be absent if e.g. NOGDI was used.
   const auto ERROR_ = ERROR;
 #undef ERROR
   const auto ERROR = ERROR_;
+#else //if macro def missing, we still need to define the symbol.
+  const auto ERROR = 0;
+#endif
 
   typedef VOID VOID_;
 #undef VOID
```

You seem to be making the assumption that we have control over the inclusion of windows.h, but this is not the case.

Please consider the following, relative to your enumerated options...

1) We don't include windows.h.  Library code we are using does (curl).
2) Irrelevant, we don't include windows.h, see (1).
3) We -don't- include windows.h in our code, see (1).

And, it is not our code, but kj/windows-sanity.h itself, that is referencing the win32::ERROR, via the 
using win32::ERROR;
when 'ERROR' was not defined when windows.h (as is apparently the case with curl's use of it.)

When the #else portion of the patch above is inactive (commented out the direct def of ERROR in #else branch), 
the version of visual studio 2019 I used complains:
```
D:\dev\tiledb\gh.tdb.pristine.20220127.git\bld.rwdi.vs19.F\externals\install\include\kj/windows-sanity.h(65): error C2039: 'ERROR': is not a member of 'win32'
D:\dev\tiledb\gh.tdb.pristine.20220127.git\bld.rwdi.vs19.F\externals\install\include\kj/windows-sanity.h(51): note: see declaration of 'win32'
D:\dev\tiledb\gh.tdb.pristine.20220127.git\bld.rwdi.vs19.F\externals\install\include\kj/windows-sanity.h(65): error C2873: 'ERROR': symbol cannot be used in a using-declaration
```

--- a/c++/src/kj/windows-sanity.h
+++ b/c++/src/kj/windows-sanity.h
@@ -48,7 +48,11 @@
 // now, we use `#pragma once` to tell the compiler never to include this file again.
 #pragma once
 
-namespace win32 {
+namespace kj_win32_workarounds {
+  // Namespace containing constant definitions intended to replace constants that are defined as
+  // macros in the Windows headers. Do not refer to this namespace directly, we'll import it into
+  // the global scope below.
+
 #ifdef ERROR  // This could be absent if e.g. NOGDI was used.
   const auto ERROR_ = ERROR;
 #undef ERROR
@@ -60,7 +64,8 @@ namespace win32 {
   typedef VOID_ VOID;
 }
 
-using win32::ERROR;
-using win32::VOID;
+// Pull our constant definitions into the global namespace -- but only if they don't already exist
+// in the global namespace.
+using namespace kj_win32_workarounds;
 
 #endif

